File: src/lib/api/exceptionStatus.ts
/**
 * 异常处理状态 API（与 makeExceptionId 对应）
 */

import { apiGet, apiPost } from '@/lib/api/client.ts'

export type ExceptionStatusRow = {
  id: string
  exception_id: string
  status: string
  updated_at: string
}

export type ExceptionStatusListResponse = {
  items: ExceptionStatusRow[]
  total: number
}

export type ExceptionStatusUpsertPayload = {
  exception_id: string
  status: 'pending' | 'ignored' | 'resolved'
}

export function listExceptionStatuses(params?: {
  exception_id?: string
  limit?: number
  offset?: number
}): Promise<ExceptionStatusListResponse> {
  const q = new URLSearchParams()
  if (params?.exception_id) q.set('exception_id', params.exception_id)
  if (params?.limit != null) q.set('limit', String(params.limit))
  if (params?.offset != null) q.set('offset', String(params.offset))
  const qs = q.toString()
  return apiGet<ExceptionStatusListResponse>(`/api/exception-statuses${qs ? `?${qs}` : ''}`)
}

export function upsertExceptionStatus(
  payload: ExceptionStatusUpsertPayload
): Promise<ExceptionStatusRow> {
  return apiPost<ExceptionStatusRow>('/api/exception-statuses', {
    exception_id: payload.exception_id,
    status: payload.status
  })
}

File: backend/app/schemas/exception_status.py
"""异常状态 API 模型。"""

from __future__ import annotations

from datetime import datetime

from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class ExceptionStatusUpsert(BaseModel):
    exception_id: str = Field(..., min_length=1)
    status: Literal["pending", "ignored", "resolved"]


class ExceptionStatusRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    exception_id: str
    status: str
    updated_at: datetime


class ExceptionStatusListResponse(BaseModel):
    items: list[ExceptionStatusRead]
    total: int
